### DISCUSSION TITLE
Changes in yarpmotorgui home positions

### DISCUSSION THREAD

**Discussion by francesco-romano** [robotology/community#187] [Announcements]
***Disclaimer***: this applies to YARP-`devel` but it will be merged into `master` in the future and released for everybody.

Dear @robotology/icub-developers  (especially @robotology/codyco-developers who make extensive use of this feature for balancing demos),

Since https://github.com/robotology/yarp/commit/af7b3e6ba5c0c69b907c8f58be621889f36e9a5a (currently in `devel`) the **`yarpmotorgui` has changed on how it handles home positions**.

The **home position** is now defined by the robot through the calibrator interfaces, and this configuration is the one applied when the user presses the `Home` button in the interface. 

A new **Custom position** which substitutes the old `zero` in the configuration files can now be specified, see [this diff](https://github.com/robotology/codyco-modules/commit/dea385fd31557b2c5f51e79fc5876154a8fac8b1#diff-af433dc8bc49fdbd0db73df4d8800386).

There is the possibility to specify up to 2 custom configurations for each configuration file, which can then be applied by the new menu option:

![screen shot 2017-02-20 at 09 49 24](https://cloud.githubusercontent.com/assets/2189180/23117840/e77019da-f751-11e6-8fd1-d9764abf2629.png)

**Comment by randaz81** [comment]
Just a quick additional note: options like `--robot --names --parts` are unnecessary and should be avoided if possible. Yarpmotorgui in devel automatically detects the available robot parts through the device description server mechanism. If the device description server is running, these options are ignored (the server has the priority on the user options)

Temporary note:
Since the development branch on icub-main is not yet merged, your files are ok for now, but please remove these options when the new yarp master release will be ready.
